src/test/java/org/opensearch/security/dlic/rest/api/AllowlistApiTest.java

```diff
@@ -0,0 +1,281 @@
+
+    private final String ENDPOINT;
+
+    public AllowlistApiTest(String endpoint){
+        ENDPOINT = endpoint;
+    }
+
+    @Parameterized.Parameters
+    public static Iterable<String> endpoints() {
+        return ImmutableList.of(
+                PLUGINS_PREFIX + "/api"
+        );
+    }
+
+    /**
+     * Helper function to test the GET and PUT endpoints.
+     *
+     * @throws Exception
+     */
+    private void testGetAndPut(final int expectedStatus, final boolean sendAdminCertificate, final Header... headers) throws Exception {
+
+        final boolean prevSendAdminCertificate = rh.sendAdminCertificate;
+        rh.sendAdminCertificate = sendAdminCertificate;
+
+        //CHECK GET REQUEST
+        response = rh.executeGetRequest(ENDPOINT + "/allowlist", headers);
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(expectedStatus));
+        if (expectedStatus == HttpStatus.SC_OK) {
+            //Note: the response has no whitespaces, so the .json file does not have whitespaces
+            Assert.assertEquals(FileHelper.loadFile("restapi/whitelist_response_success.json"), FileHelper.loadFile("restapi/whitelist_response_success.json"));
+        }
+        //FORBIDDEN FOR NON SUPER ADMIN
+        if (expectedStatus == HttpStatus.SC_FORBIDDEN) {
+            assertTrue(response.getBody().contains("API allowed only for super admin."));
+        }
+        //CHECK PUT REQUEST
+        response = rh.executePutRequest(ENDPOINT + "/allowlist", "{\"enabled\": true, \"requests\": {\"/_cat/nodes\": [\"GET\"],\"/_cat/indices\": [\"GET\"] }}", headers);
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(expectedStatus));
+
+        rh.sendAdminCertificate = prevSendAdminCertificate;
+    }
+
+    /**
+     * Tests that the response does not have a _meta header
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testResponseDoesNotContainMetaHeader() throws Exception {
+
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        RestHelper.HttpResponse response = rh.executeGetRequest(ENDPOINT + "/allowlist");
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        Assert.assertFalse(response.getBody().contains("_meta"));
+    }
+
+    /**
+     * Tests that putting an unknown key fails
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testPutUnknownKey() throws Exception {
+
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        RestHelper.HttpResponse response = rh.executePutRequest(ENDPOINT + "/allowlist", "{ \"unknownkey\": true, \"requests\": {\"/_cat/nodes\": [\"GET\"],\"/_cat/indices\": [\"GET\"] }}");
+        Assert.assertEquals(HttpStatus.SC_BAD_REQUEST, response.getStatusCode());
+        assertTrue(response.getBody().contains("invalid_keys"));
+        assertHealthy();
```

Comment:
If you check all the tests, only 2 of them have this check. 
Adding the check to all tests would unnecessarily increase the test run time unless there's a good reason to do that.

Reply:
> good reason to do that.

If we added a bug where the service was destabilized after one of these calls, wouldn't that be worth catching?  If you'd like we can revisit this topic later.  As it seems like the service should start in a health state and also end in a healthy state for all of our tests cases.

Reply:
We can revisit this topic later. This isn't specific to this particular test class.

Reply:
Can you create an issue to track this?

Reply:
I don't even see the why it's needed to do health check at the end of all tests. We can discuss this out of this PR. If it's this is a necessary enhancement, we can create an issue later. Otherwise, the issue would be opened and closed.